stl/inc/atomic

```diff
@@ -462,28 +475,30 @@ struct _Atomic_storage {
         const auto _Storage_ptr  = _STD addressof(_Storage);
         const auto _Expected_ptr = _STD addressof(_Expected);
 
-        for (;;) {
+        auto _Are_equal = [this, _Order](const void* _Comparand) -> bool {

```

Comment:
I think the `-> bool` is unnecessary (all of the `return` statements are `bool`) although if you want to keep it, I won't object.

Reply:
Removed